tests/01-known-values.sh

```diff
@@ -3,25 +3,22 @@
 ### Constants
 # The scenario command requires a lot of memory, so valgrind is only enabled
 # if $USE_VALGRIND > 1.
-scenario_valgrind_min=2
+c_valgrind_min=2
 
 ### Actual command
 scenario_cmd() {
-	$val_cmd $bindir/tests/test_scrypt \
-		1> $out/$basename-stdout.txt
-	cmd_retval=$?
-
-	echo "$cmd_retval"
-}
-
-### Check output
-scenario_check() {
-	retval=$1
+	# Run the binary which tests known input/output strings.
+	setup_check_variables
+	(							\
+		$c_valgrind_cmd $bindir/tests/test_scrypt	\
+			1> $s_basename-stdout.txt		\
+	; echo $? > $c_exitfile )
```

Comment:
BTW if you want to allow ${cmd} to complete later, you probably want to add a `&` after the `)` ending the subshell?